libraries/stdlib/jvm/test/random/RandomSerializationTest.kt

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package random
+
+import test.io.deserializeFromHex
+import test.io.serializeAndDeserialize
+import kotlin.random.Random
+import kotlin.random.asJavaRandom
+import kotlin.random.asKotlinRandom
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertSame
+import kotlin.test.assertTrue
+
+class RandomSerializationTest {
+    @Test
+    fun defaultIsSerializable() {
+        val instance = Random
+        discardSomeValues(instance)
+        assertSame(instance, serializeAndDeserialize(instance))
+    }
+
+    private fun discardSomeValues(instance: Random) {
+        instance.nextInt()
+        instance.nextDouble()
+        instance.nextLong()
+        instance.nextBytes(64)
+    }
+
+    private fun testRandomsHaveSameState(first: Random, second: Random) {
+        assertEquals(first.nextInt(), second.nextInt())
+        assertEquals(first.nextDouble(), second.nextDouble())
+        assertEquals(first.nextLong(), second.nextLong())
+        assertTrue(first.nextBytes(64).contentEquals(second.nextBytes(64)))
+    }
+
+    @Test
+    fun deserializeDefault() = assertSame(
+        expected = Random,
+        actual = deserializeFromHex("ac ed 00 05 73 72 00 1c 6b 6f 74 6c 69 6e 2e 72 61 6e 64 6f 6d 2e 52 61 6e 64 6f 6d 24 44 65 66 61 75 6c 74 59 81 49 e9 14 4e a8 28 02 00 00 78 70")
```

Comment:
Please also add a similar test for XorwowRandom.

Reply:
OK

Reply:
It is impossible since serialization produces a new instance with state restored from bytes, so instances of XorWow will be neither same nor equal (since XorWow is not a data class)

Reply:
Indeed, that's why I said "a similar", not "the same" :)
You can use `testRandomsHaveSameState` to check that the state has been restored correctly.

Reply:
OK